Quotient/keyimport.cpp

```diff
@@ -87,8 +90,65 @@ KeyImport::Error KeyImport::importKeys(QString data, const QString& passphrase,
         room->addMegolmSessionFromBackup(
             keyObject["session_id"_ls].toString().toLatin1(),
             keyObject["session_key"_ls].toString().toLatin1(), 0,
-            keyObject[SenderKeyKey].toVariant().toByteArray());
+            keyObject[SenderKeyKey].toVariant().toByteArray(),
+            keyObject["sender_claimed_keys"_ls]["ed25519"_ls].toString().toLatin1()
+        );
     }
     return Success;
 }
 
+Quotient::Expected<QByteArray, KeyImport::Error> KeyImport::encrypt(QJsonArray sessions, const QString& passphrase)
+{
+    auto plainText = QJsonDocument(sessions).toJson(QJsonDocument::Compact);
+
+    auto salt = getRandom(AesBlockSize);
+    auto iv = getRandom(AesBlockSize);
+    quint32 rounds = 200'000; // spec: "N should be at least 100,000";
+
+    auto keys = pbkdf2HmacSha512<64>(passphrase.toLatin1(), salt.viewAsByteArray(), rounds);
+    if (!keys.has_value()) {
+        qCWarning(E2EE) << "Failed to calculate pbkdf:" << keys.error();
+        return OtherError;
+    }
+
+    auto result = aesCtr256Encrypt(plainText, byte_view_t<Aes256KeySize>(keys.value().begin(), Aes256KeySize), asCBytes<AesBlockSize>(iv.viewAsByteArray()));
+
+    if (!result.has_value()) {
+        qCWarning(E2EE) << "Failed to encrypt export" << result.error();
+        return OtherError;
+    }
+
+    QByteArray data;
+    data.append("\x01");
+    data.append(salt.viewAsByteArray());
+    data.append(iv.viewAsByteArray());
+    QByteArray roundsData(4, u'\x0');
+    qToBigEndian<quint32>(rounds, roundsData.data());
+    data.append(roundsData);
+    data.append(result.value());
+    auto mac = hmacSha256(key_view_t(keys.value().begin() + 32, 32), data);
+    if (!mac.has_value()) {
+        qCWarning(E2EE) << "Failed to calculate MAC" << mac.error();
+        return OtherError;
+    }
+    data.append(mac.value());
+
+       return "-----BEGIN MEGOLM SESSION DATA-----\n"_ba
+           % (std::views::chunk(data.toBase64(), 96) | std::views::join_with('\n')
+              | std::ranges::to<QByteArray>())
+           % "\n-----END MEGOLM SESSION DATA-----\n"_ba;
```

Comment:
Ok, here's a workaround while we wait for `std::ranges::to`:
```suggestion
   return "-----BEGIN MEGOLM SESSION DATA-----\n"_ba
          % std::ranges::fold_left(std::views::chunk(data.toBase64(), 96)
                                       | std::views::join_with('\n'),
                                   QByteArray{}, std::plus<>())
          % "\n-----END MEGOLM SESSION DATA-----\n"_ba;
```